### ISSUE TITLE
Outputs from multiple threads are interleaved

### ISSUE THREAD

**Issue by snathan9** [ansible/ansible#8460]
##### Issue Type:

Bug Report
##### Ansible Version:

1.6.10
##### Environment:

Linux ubrts1 3.8.0-34-generic #49~precise1-Ubuntu
##### Summary:

On running the "setup" module to validate a large group of servers (grouped as sg1 in e.g. below), the output gets interleaved and unable to get the correct individual server specifications.
##### Steps To Reproduce:

ansible sg1 -m setup > sg1.txt

Workaround is to limit fork to 1 which limits the effectiveness of the tools.

ansible sg1 -m setup -f 1 > sg1.txt
##### Expected Results:

The expected result was each individual server output to not be interwoven with any other server output. 
##### Actual Results:

The actual output has data from multiple servers intewoven together making it impossible to determine the individual server specification, which is the purpose of the command.

**Comment by mpdehaan** [comment]
Can I ask if there's anything interesting about your host OS or filesystem?

We have locking to avoid interleaving and this has been a thing for some time, so I'm wondering if there's an NFS dir or some slightly unusual OS choices afoot?

**Comment by snathan9** [comment]
##### Machine:

The machine is a Virtual Machine running on VirtualBox.  The Host is a Windows 7 Workstation and the Guest is the Ubuntu server on which ansible is installed.

##### File system:

No NFS, just local disks on the VM itself.

##### Network:

The underlying systems that "setup" runs on are remote servers so they may not all respond at the same rate and may have some latency, however there is no connectivity issues for sure.

##### Target  OS:

The target machines are all Redhat 6.4 on a data center and virtualized with Vmware.

Hope that helps.

**Comment by mpdehaan** [comment]
Can I ask if how much in VirtualBox might be a shared with the Windows guest?

I'm guessing this is the problem as we're using "flock()" and that requires something Linuxey, which would explain the problem.

The solution would be to share less of the filesystem.

**Comment by snathan9** [comment]
The filesystem where the output goes is all within Linux.  Not using anything windows (such as samba) or anything VirutalBox (such as shared folders with host).  It's all inside Linux VM. So to answer your question there is shared nothing except of course the guest resources for VM via VirtualBox hypervisor.

I am inclined to believe this is more a native ansible issue, but again you never know :)

**Comment by snathan9** [comment]
Installed on Redhat 6.4 on bare metal and have the same behavior.  It's quite easy to reproduce as it happens consistently.

**Comment by koenigma1** [comment]
I observed this issue on MAC OS X with ansible version 1.7.1.

**Comment by mpdehaan** [comment]
VirtualBox shared FS could be causing the issue here because it can't get an flock() on it's weird sharing filesystem for the above ticket from @snathan9 

I'm not sure about you @koenigma1, any ideas?

**Comment by snathan9** [comment]
The behavior has been reproduced in "Redhat 6.4 on bare metal".  Please refer to earlier comment on Aug 11th.

**Comment by tomek-kuchta** [comment]
Hello, I also see some interleaving on Ansible 1.9.4 logging. I use VirtualBox share, but the log is stored under /tmp directory of the virtual machine, which is not shared. I run Ansible on 16 LXC containers which are also under the same VM.

**Comment by jctanner** [comment]
Hi, our apologies on taking so long to follow-up on this issue.

This bug was reported on a version of Ansible that is very out of date, and we're not sure if it's valid anymore. As such, could you please let us know if this issue still applies to the devel branch, or the most recent version of Ansible (currently 2.1.x or the latest commit from the stable-2.1 branch)?

If we do not see any follow-up on this issue, we will close and lock this issue and any further reports should be done in a new issue.

Thanks!

**Comment by ansibot** [comment]
@snathan9 ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@snathan9 ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by bcoca** [comment]
Closing for lack of response and because this is an old unsupported version.

If you continue seeing any problems related to this issue on a current version, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:

https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code
Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
